pkg/customresourcestate/generate/cmd.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2022 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generate
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/controller-tools/pkg/genall"
+	"sigs.k8s.io/controller-tools/pkg/genall/help"
+	prettyhelp "sigs.k8s.io/controller-tools/pkg/genall/help/pretty"
+	"sigs.k8s.io/controller-tools/pkg/loader"
+	"sigs.k8s.io/controller-tools/pkg/markers"
+
+	"k8s.io/kube-state-metrics/v2/pkg/customresourcestate/generate/generator"
+)
+
+const (
+	generatorName = "metric"
+)
+
+var (
+	// optionsRegistry contains all the marker definitions used to process command line options
+	optionsRegistry = &markers.Registry{}
+
+	generateWhichMarkersFlag bool
+)
+
+// GenerateCommand runs the kube-state-metrics custom resource config generator.
+var GenerateCommand = &cobra.Command{
+	Use:                   "generate [flags] /path/to/package [/path/to/package]",
+	Short:                 "Generate custom resource metrics configuration from go-code markers (experimental).",
+	DisableFlagsInUseLine: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if generateWhichMarkersFlag {
+			PrintMarkerDocs()
+			return nil
+		}
+
+		if len(args) == 0 {
+			return fmt.Errorf("requires at least 1 arg(s), only received %d", len(args))
```

Comment:
Do we really need to look into len(args), if we're only erroring on zero args?

Reply:
Adjusted 👍